src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -176,4 +176,23 @@ public class Memory {
         CurrentlyWritingByte = value;
         _writeBreakPoints.TriggerMatchingBreakPoints(address);
     }
+
+    // Overloads to work with segment:offset addresses.
+    public uint GetUint32(ushort segment, ushort offset) => 
+        GetUint32((uint)((segment << 4) + offset));
```

Comment:
Use ToPhysicalAddress.ToPhysicalAddress to compute the actual physical address.

Reply:
Oops, hadn't noticed those. Will do.